src/Messenger/KafkaSender.php

```diff
@@ -18,18 +19,20 @@ class KafkaSender implements SenderInterface
     private SerializerInterface $serializer;
     private RdKafkaFactory $rdKafkaFactory;
     private KafkaSenderProperties $properties;
-    private KafkaProducer $producer;
+    private ?KafkaProducer $producer = null;
 
     public function __construct(
         LoggerInterface $logger,
         SerializerInterface $serializer,
         RdKafkaFactory $rdKafkaFactory,
-        KafkaSenderProperties $properties
+        KafkaSenderProperties $properties,
+        ?FlushOnTerminate $flushOnTerminate
     ) {
         $this->logger = $logger;
         $this->serializer = $serializer;
         $this->rdKafkaFactory = $rdKafkaFactory;
         $this->properties = $properties;
+        $this->flushOnTerminate = $flushOnTerminate;
```

Comment:
it looks like this property not defined in class, dynamic property creation is bad